### PR TITLE
Mes-5800: implement auth connect

### PR DESCRIPTION
**Pull request by andrewsetterfield**

### CODE REVIEW

**Review by szabikr — APPROVED** [review]
Looks good! Will be a great base for the Template project. Highlighting the TODOs in the README so that devs who use the template know what are the additional things that they need to do in order to get started would be a useful addition to this PR

File: src/app/shared/classes/base-page.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  protected constructor(
+    public platform: Platform,
+    public authenticationProvider: AuthenticationProvider,
+    public router: Router,
+    public loginRequired: boolean = true,
+  ) {
+
+  }
+
+  ionViewWillEnter() {
```
Comment (Cullima) [inline src/app/shared/classes/base-page.ts:16]: shouldn't this sort of thing be covered in the route guard now? that's essentially what this is

File: src/app/shared/classes/logout-base-page.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { BasePageComponent } from './base-page';
+import { AlertController, Platform } from '@ionic/angular';
+import { AuthenticationProvider } from '../../providers/authentication/authentication';
+import { Router } from '@angular/router';
+
+export abstract class LogoutBasePageComponent extends BasePageComponent {
```
Comment (Cullima) [inline src/app/shared/classes/logout-base-page.ts:6]: not sure what the purpose of this new abstract class is, or why it's needed

Comment (andrewsetterfield) [reply src/app/shared/classes/logout-base-page.ts:6]: its was how current app implements the logout functionality on any page that needed it.  Can be moved to a common module

Comment (andrewsetterfield) [reply src/app/shared/classes/logout-base-page.ts]: moved this into a provider

**Review by Cullima — APPROVED** [review]
Looks good!

**Pull request merged**